src/rulesets/SparseArrays/array.jl

```diff
@@ -0,0 +1,16 @@
+using SparseArrays
+
+function rrule(::Type{<:SparseMatrixCSC{T,N}}, arr) where {T,N}
+  function SparseMatrix_pullback(Δ)
+    (collect(Δ),)
+  end
+  SparseMatrixCSC{T,N}(arr), SparseMatrix_pullback
+end
+
+function rrlue(::typeof(diagm), x)
+  diagm(x), d -> (diag(d),)
+end
```

Comment:
`diagm` is defined in LinearAlegebra stdlib, not SpareArrays, so rules for it should go in `rulesets/LinearAlgebra/dense.jl`. Since there are other `rrules` for `diagm`, I think we should probably restrict the input to `v::AbstractVector`?

also, please can you change this to follow the ChainRules style of using named local functions for the pullback in an rrule?
https://www.juliadiff.org/ChainRulesCore.jl/dev/writing_good_rules.html#Code-Style

Reply:
I've moved things, good to get another pair of eyes over it